pkg/resourcekeeper/gc.go

```diff
@@ -245,23 +245,71 @@ func (h *gcHandler) Sweep(ctx context.Context) (finished bool, waiting []v1beta1
 }
 
 func (h *gcHandler) recycleResourceTracker(ctx context.Context, rt *v1beta1.ResourceTracker) error {
```

Comment:
can we make the recycle functions to work per component? That can be more clear like:

```
func recycleComponents(compnents ..){
    for _, comp:=range components{
     if checkDependsOnComponents(comp){
         deleteDependsOnComponents(comp)
     }
    delete(comp)
   }
}
```

Reply:
The function `deleteDependsOnComponents`  also will be a iterate function to check the relation of dependsOn. And current gc is not able to recycle at the compnent level.